enh(administration): add chapter how to secure a Centreon platform (#173)
centreon/centreon-documentation: en/administration/secure-platform.md

        RewriteRule ^index\.html$ - [L]
        RewriteCond %{REQUEST_FILENAME} !-f
        RewriteCond %{REQUEST_FILENAME} !-d
        RewriteRule . /index.html [L]
        ErrorDocument 404 /centreon/index.html

        AddType text/plain hbs
    </Directory>

    <Directory "/usr/share/centreon/api">
        Options Indexes
        AllowOverride all
        Order allow,deny
        Allow from all
        Require all granted
        <IfModule mod_php5.c>
            php_admin_value engine Off
        </IfModule>

        AddType text/plain hbs
    </Directory>
</VirtualHost>

RedirectMatch ^/$ /centreon
```

> Don't forget to change **SSLCertificateFile** and **SSLCertificateKeyFile** directives with the path containing your certificate and key.

5. Enable HttpOnly and Secure flags

Edit the **/opt/rh/httpd24/root/etc/httpd/conf/httpd.conf** file and add the following line:
```apacheconf
Header always edit Set-Cookie ^(.*)$ $1;HttpOnly;Secure
```

6. Restart the Apache to take in account the new configuration:

```shell
systemctl restart httpd24-httpd
```

Then check its status:

```shell
systemctl status httpd24-httpd
```

If everything is ok, you must have:

> ```shel

Reviewer: ```suggestion
```shell
```

---
id: secure-platform
title: Secure your platform
---

This chapter suggests how to best secure your Centreon platform.

## Strengthen user account security

After installing Centreon, it is necessary to change the default passwords of the following users:

- root
- centreon
- centreon-engine
- centreon-broker
- centreon-gorgoned

To do this, use the following command from **root** account:

```shell

> password <account_name>

Reviewer: isn't passwd ?
```suggestion
passwd <account_name>
```

```

In addition, it is important to verify that the Apache account does not have connection rights to the terminal.
Execute the following command:
```shell
cat /etc/passwd | grep apache
```

You must have **/sbin/nologin** like:
```shell
apache:x:48:48:Apache:/usr/share/httpd:/sbin/nologin
```

> As a reminder, the list of users and groups can be found [here](../installation/prerequisites.html#users-and-groups)

## Securing the installation of the DBMS

[MariaDB](https://mariadb.com/kb/en/mysql_secure_installation/) propose a default procedure to securize the DBMS
installation. Please execute the following command and follow instruction:
```shell
mysql_secure_installation
```

## Securing the Apache web server

By default, Centreon installs a web server in HTTP mode. It is strongly recommended to switch to HTTPS mode by adding your certificate.

> It is better to use a certificate validated by an authority rather than to use a self-signed certificate.

Reviewer: ```suggestion
It is better to use a certificate validated by an authority rather than a self-signed one.
```

If you do not have a certificate validated by an authority, you can generate one on platforms such as [Let's Encrypt](https://letsencrypt.org/).

> Once your certificate obtained, perform the following procedure to activate HTTPS mode on your Apache server:

Reviewer: ```suggestion
Once you obtained a certificate, perform the following procedure to activate HTTPS mode on your Apache server:
```

    <Directory "/usr/share/centreon/www">
        DirectoryIndex index.php
        Options Indexes
        AllowOverride all
        Order allow,deny
        Allow from all
        Require all granted
        <IfModule mod_php5.c>
            php_admin_value engine Off
        </IfModule>

        RewriteRule ^index\.html$ - [L]
        RewriteCond %{REQUEST_FILENAME} !-f
        RewriteCond %{REQUEST_FILENAME} !-d
        RewriteRule . /index.html [L]
        ErrorDocument 404 /centreon/index.html

        AddType text/plain hbs
    </Directory>

    <Directory "/usr/share/centreon/api">
        Options Indexes
        AllowOverride all
        Order allow,deny
        Allow from all
        Require all granted
        <IfModule mod_php5.c>
            php_admin_value engine Off
        </IfModule>

        AddType text/plain hbs
    </Directory>
</VirtualHost>

RedirectMatch ^/$ /centreon
```

> Don't forget to change **SSLCertificateFile** and **SSLCertificateKeyFile** directives with the path containing your certificate and key.

5. Enable HttpOnly and Secure flags

Edit the **/opt/rh/httpd24/root/etc/httpd/conf/httpd.conf** file and add the following line:

> Header always edit Set-Cookie ^(.*)$ $1;HttpOnly;Secure

Reviewer: là j'ai pas compris

           ├─31903 /opt/rh/httpd24/root/usr/sbin/httpd -DFOREGROUND
           └─32050 /opt/rh/httpd24/root/usr/sbin/httpd -DFOREGROUND
```

## Custom URI

It is possible to update the URI of Centreon. For example, **/centreon** can be
replaced by **/monitoring**.

To update the Centreon URI, you need to follow those steps:

1. Go to **Administration > Parameters > Centreon UI** and change the **Centreon Web Directory** value.

![image](../assets/administration/custom-uri.png)

2. Edit Apache configuration file for Centreon Web (**/opt/rh/httpd24/root/etc/httpd/conf.d/10-centreon.conf**) and
change **/centreon** path with your new path

## User authentication

Centreon offers several methods to authenticate users:

- local (MySQL)
- [LDAP](./parameters/ldap.html)
- [Generic SSO](./parameters/centreon-ui.html#sso) or [Keycloak SSO](./parameters/centreon-ui.html#keycloak-sso)

## Create user profiles

Centreon offers to manage access permissions to the different menus, resources and possible actions on resources via
the management of [Access Control List](./access-control-lists.html).

## Secure communications between servers

> It strongly recomanded to secure communications between the different servers of the Centreon platform if some servers

Reviewer: ```suggestion
It is strongly recommended to secure communications between the different servers of the Centreon platform if some servers
```

> are not in secure network.

Reviewer: ```suggestion
are not in a secure network.
```

## Secure communications between servers

> It strongly recommended to secure communications between the different servers of the Centreon platform if some servers

Reviewer: ```suggestion
It is strongly recommended to secure communications between the different servers of the Centreon platform if some servers
```
